server.go

```diff
@@ -2687,19 +2687,18 @@ func (s *server) ConnectToPeer(addr *lnwire.NetAddress, perm bool) error {
 
 	// Peer was not found, continue to pursue connection with peer.
 
-	// If there's already a pending connection request for this pubkey,
-	// then we ignore this request to ensure we don't create a redundant
-	// connection.
-	if reqs, ok := s.persistentConnReqs[targetPub]; ok {
-		srvrLog.Warnf("Already have %d persistent connection "+
-			"requests for %v, connecting anyway.", len(reqs), addr)
-	}
-
-	// If there's not already a pending or active connection to this node,
-	// then instruct the connection manager to attempt to establish a
-	// persistent connection to the peer.
 	srvrLog.Debugf("Connecting to %v", addr)
 	if perm {
+
+		// If there's already a pending connection request for this pubkey,
+		// then we cancel pending request to ensure we don't create a redundant
+		// connection.
+		if reqs, ok := s.persistentConnReqs[targetPub]; ok {
+			srvrLog.Warnf("Already have %d persistent connection "+
+				"requests for %v, connecting anyway.", len(reqs), addr)
+			s.cancelConnReqs(pubStr, nil)
```

Comment:
imo we don't need to do this (sorry if the old comment was confusing), it should be fine to have multiple pending connection requests to the same peer. note that these may actually be at different addresses, even though they have the same pubkey, and we will use whichever succeeds first

Reply:
@cfromknecht thanks for your review. I agree it creates a problem.
Originally I wanted to solve the case where two requests simultaneously tries  to connect but somehow they prevent from each other to connect.
The behavior I experienced is a constantly connect/disconnect effect even though the code aims to handle this case.
Could be that the behavior I experienced is related to a problem that might be solved by this PR: https://github.com/btcsuite/btcd/pull/1330
without the above PR connections are not cancelled if they are waiting for retry. so it could be that the cancelation of other requests when one succeed didn't work properly.

Anyway, we both agree the solution is not to cancel at this point and I will need to repro that behavior so I can suggest a better solution.